scapy/layers/dns.py

```diff
@@ -276,14 +283,30 @@ def mysummary(self):
                 name = ' "%s"' % self.qd.qname
         return 'DNS %s%s ' % (type, name)
 
-dnstypes = { 0:"ANY", 255:"ALL",
-             1:"A", 2:"NS", 3:"MD", 4:"MF", 5:"CNAME", 6:"SOA", 7: "MB", 8:"MG",
-             9:"MR",10:"NULL",11:"WKS",12:"PTR",13:"HINFO",14:"MINFO",15:"MX",16:"TXT",
-             17:"RP",18:"AFSDB",28:"AAAA", 33:"SRV",38:"A6",39:"DNAME",
-             41:"OPT", 43:"DS", 46:"RRSIG", 47:"NSEC", 48:"DNSKEY",
-	     50: "NSEC3", 51: "NSEC3PARAM", 32769:"DLV" }
+    def post_build(self, pkt, pay):
+        if isinstance(self.underlayer, TCP) and self.length is None:
+            pkt = struct.pack("!H", len(pkt) - 2) + pkt[2:]
+        return pkt + pay
+
+
+# https://en.wikipedia.org/wiki/List_of_DNS_record_types
```

Comment:
Actually, 255 was suppressed but is in `dnsqtypes` which is used to `.update()` `dnstypes`. I only removed the duplicate entry. Anyway, link updated since yours is better.